internal/terraform/context.go

```diff
@@ -42,6 +43,7 @@ type ContextOpts struct {
 	Parallelism  int
 	Providers    map[addrs.Provider]providers.Factory
 	Provisioners map[string]provisioners.Factory
+	Backends     func(string) backend.InitFn
```

Comment:
Question to reviewer - is this approach touching too much of the codebase?

ContextOps is used in a variety of places beyond `terraform test`. One of the upsides of my previouas, alternative approach (https://github.com/hashicorp/terraform/pull/36646/commits/4afc3d79e38c46fccd292d8169d50bb97a309dec) was that it was test-specific.

On the other hand, adding another factory here matches what's already in place for providers and provisioners.

Reply:
If however we decide to use the ContextOpts, we definitely need some comment here indicating that it only related to terraform test

Reply:
Agreed re: the comment!

I'll swap back to the original approach. I like that isn't test specific, and even though it requires passing a value between multiple layers in the code it has the benefit of being explicit, and something easy to refactor in future.